crates/rspack_core/src/compiler/compilation.rs

```diff
@@ -1306,6 +1306,13 @@ impl Compilation {
       Ok(())
     }
 
+    // FIXME:
+    // Webpack may modify the moduleGraph in module.getExportsType()
+    // and it is widely called after compilation.finish()
+    // so add this method to trigger moduleGraph modification and
+    // then make sure that moduleGraph is immutable
+    prepare_get_exports_type(&mut self.module_graph);
```

Comment:
Trigger getMaxTarget and getExportInfoId before code generation to make sure that module graph is immutable during code generation